Add basic styles to about:about and created page w/ version information (about:brave)

bsclifton opened the pull request:
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Add basic styles and version information to about:about

Fixes https://github.com/brave/browser-laptop/issues/463

Auditors: @bbondy, @bradleyrichter

## Test Plan
1. Launch Brave and visit about:about
2. Pull up versions from the "About Brave" message box: Help > About Brave (or on Mac, Brave > About Brave)
3. Compare the versions shown on the about:about page to the ones in the message box

## Screenshots
**The new look**
![screen shot 2016-11-06 at 2 20 12 am](https://cloud.githubusercontent.com/assets/4733304/20036603/6187747a-a3c9-11e6-8790-174a9165ed48.png)

**Text is really easy to select for copy/paste**
![screen shot 2016-11-06 at 2 22 27 am](https://cloud.githubusercontent.com/assets/4733304/20036607/7563c872-a3c9-11e6-9950-74edbc8942bf.png)

**compare this to the current look**
![screen shot 2016-11-06 at 2 35 14 am](https://cloud.githubusercontent.com/assets/4733304/20036620/c44931c0-a3c9-11e6-9246-8ea2af68ebe0.png)

bbondy commented on js/about/about.js:
```diff
@@ -3,23 +3,63 @@
+    ipc.on(messages.VERSION_INFORMATION_UPDATED, (e, versionInformation) => {
+      console.log('got message')
+      if (this.state.versionInformation.size === 0) {
+        console.log('updating')
```
nit: logging

bbondy commented:
This is great, I don't mind about:about being combined either. Can you get rid of the dialog though from the about menu and make it go here instead? Test main menu locations and hamburger menu location.

bsclifton commented:
Maybe once there's more about pages, we can break this out into it's own about:version page; my initial thought was that (with limited info) it makes sense to show as many details about Brave as possible in one place :smile: We could possibly add more info we find valuable for troubleshooting, like OS version, Video drivers, etc.

Working on the fixes now...

luixxiul commented:
Wouldn't it nice if we display the additional version info like "RC1" or "preview1"?

bsclifton commented:
Good call, @luixxiul!  I captured that with https://github.com/brave/browser-laptop/issues/5462

bsclifton commented:
Updated! @darkdh, if you have a moment, please check out the changes. There are two message boxes that were in `app/aboutDialog.js` which I moved into `app/importer.js`. They were being exported, but I changed them to local `const` because they're not used elsewhere.  I tested an import to verify it loads (but didn't try the one w/ a warning).

@bbondy picking `About Brave` now brings you to the about page :smile: The previous message box has been removed

bsclifton commented:
Here we go- even more useful info for debugging :smile: 

![screen shot 2016-11-07 at 11 34 30 am](https://cloud.githubusercontent.com/assets/4733304/20070589/39ed522a-a4de-11e6-95fd-c31f0dd67016.png)

darkdh commented:
++ for the importer dialog, thx.

bbondy commented:
cc @bradleyrichter do we want to combine the about page with about:about? Note other browsers keep them separate and we might double or triple our about:about pages in the next year. I'm fine either way.

bsclifton commented on js/about/entry.js:
```diff
@@ -6,50 +6,54 @@ const ipc = window.chrome.ipc
+    break
   case 'about:styles':
     element = require('./styles')
     break
```
this is sorted alphabetical now

bsclifton commented on js/lib/appUrlUtil.js:
```diff
@@ -52,21 +52,22 @@ module.exports.getManifestUrl = function () {
-  'about:flash': module.exports.getAppUrl('about-flash.html'),
-  'about:error': module.exports.getAppUrl('about-error.html'),
-  'about:autofill': module.exports.getAppUrl('about-autofill.html'),
   'about:styles': module.exports.getAppUrl('about-styles.html')
```
same here; alphabetical FTW :stuck_out_tongue:

bbondy commented:
fancy with the sortable table 👍

bbondy commented:
fannnntastic, thanks.

srirambv commented:
@bsclifton doesn't brave/muon/channel/os needs to be capitalized??

bsclifton commented:
@srirambv that is a good question! I didn't consider that- I'll do a follow up which fixes that :smile:
